### PR TITLE
Add HTTP circuit breaker with retry test case

### PR DESCRIPTION
**Pull request by ThisaruGuruge**
## Security checks
 - Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? yes
 - Ran FindSecurityBugs plugin and verified report? yes
 - Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets? yes

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -133,6 +133,16 @@
...
         </profile>
+        <profile>
+            <id>http-resiliency-circuit-breaker-with-retry</id>
+            <modules>
+                <module>scenarios-commons</module>
+                <module>http/resiliency</module>
+            </modules>
+            <activation>
+                <activeByDefault>true</activeByDefault>
+            </activation>
```
Comment (Manuri) [inline pom.xml]: There is no requirement to make this active by default I think.

File: .testgrid-http-resiliency.yaml [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# TestGrid Test Configuration Parameters
+version: '0.9'
+emailToList: "thisaru@wso2.com"
```
Comment (Manuri) [inline .testgrid-http-resiliency.yaml]: We should add a common mailing list.

File: http/resiliency/src/test/java/org/ballerinalang/scenario/test/http/resiliency/CircuitBreakerWithRetryTest.java [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+    }
+
+    @Test(description = "Circuit breaker with retry test - http1 to http1")
+    public void testCircuitBreakerWithRetryHttp1() throws IOException {
+        String message = "[Http1Service] OK response. Backend request count: 3 Circuit breaker request count: 5 " +
+                "Retry request count: 1";
+        String url = "http://" + host + ":" + port + "/getResponse";
+        HttpResponse httpResponse = HttpClientRequest.doGet(url);
+        verifyResponse(httpResponse, message, 200);
+    }
```
Comment (Manuri) [inline http/resiliency/src/test/java/org/ballerinalang/scenario/test/http/resiliency/CircuitBreakerWithRetryTest.java]: Do we have any other test scenarios (may be error cases etc) that we need to add here?

File: http/resiliency/src/test/resources/source_files/circuitbreaker-with-retry-test/src/retry_service/http1_retry.bal [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+    @http:ResourceConfig {
+        methods: ["GET"]
+	}
+    resource function getResponse(http:Caller caller, http:Request request) {
+        count += 1;
+        log:printInfo(servicePrefix + "Request Received. Request Count: " + count.toString());
+        var backendResponse = backendClientEP->forward("/getResponse", request);
+        http:Response response = new;
+        if (backendResponse is http:ClientError) {
+            io:println(backendResponse);
```
Comment (Manuri) [inline http/resiliency/src/test/resources/source_files/circuitbreaker-with-retry-test/src/retry_service/http1_retry.bal]: Do we need the print statements?

Comment (ThisaruGuruge) [reply http/resiliency/src/test/resources/source_files/circuitbreaker-with-retry-test/src/retry_service/http1_retry.bal]: Will remove them

File: http/resiliency/src/test/java/org/ballerinalang/scenario/test/http/resiliency/CircuitBreakerWithRetryTest.java (hunk 3 above)
Comment (ThisaruGuruge) [reply http/resiliency/src/test/java/org/ballerinalang/scenario/test/http/resiliency/CircuitBreakerWithRetryTest.java]: Yes there should be more tests. Will add them once this is running properly.

File: .testgrid-http-resiliency.yaml (hunk 2 above)
Comment (ThisaruGuruge) [reply .testgrid-http-resiliency.yaml]: Done

**Pull request merged**